src/main/webapp/profileScript.js

```diff
@@ -107,13 +108,37 @@ function addRoute(newRoute) {
   routeRating.classList.add("rating");
 
   routeDetails.innerHTML = newRoute["routeName"];
-  let emptyStart = '<span class="far fa-star"></span>';
-  routeRating.innerHTML = emptyStart.repeat(5);
+  let emptyStar = '<span class="far fa-star"></span>';
+  let halfStar = '<span class="fas fa-star-half-alt"></span>';
+  let fullStar = '<span class="fas fa-star checked"></span>';
+
+  let ratingCopy = newRoute["rating"];
+  let numberOfFullStars = 0;
+  while (ratingCopy >= 1) {
+    numberOfFullStars += 1;
+    ratingCopy -= 1;
+  }
+  let numberOfHalfStars = 0;
+  if (ratingCopy > 0) {
+    numberOfHalfStars = 1;
+  }
+
+  routeRating.innerHTML =
+    fullStar.repeat(numberOfFullStars) +
+    halfStar.repeat(numberOfHalfStars) +
+    emptyStar.repeat(5 - numberOfFullStars - numberOfHalfStars);
 
   card.appendChild(container);
   container.appendChild(routeDetails);
   container.appendChild(routeImg);
   container.appendChild(routeRating);
 
-  document.getElementById("completed-routes").appendChild(card);
+  // 2 - user is editor, 1 - user is owner
+  if (newRoute["type"] == 2) {
```

Comment:
I think it's better to use enums in this case
https://www.sohamkamani.com/blog/2017/08/21/enums-in-javascript/

Because it is very easy to make mistakes, when you use numbers to describe states.